00 Core/MWSE/mods/tew/AURA/volumeSave.lua

```diff
@@ -164,57 +196,79 @@ end
 local function doModules()
     local menu = tes3ui.findMenu(this.id_menu)
     local mp = tes3.mobilePlayer
-    for moduleName, data in pairs(moduleData) do
-        if mp and data.new and data.newRef and tes3.getSoundPlaying{sound = data.new, reference = data.newRef} then
-            local ref = data.newRef
-            local track = data.new
-            local configKey = common.getInteriorType(this.cell):gsub("ten", "sma")
-            local sc = {}
-            local entry = createEntry()
-            local trackInfo = entry:findChild(this.id_trackInfo)
-
-            -- No point adjusting tracks attached to refs other than player's while underwater
-            if cellData.playerUnderwater and (ref ~= mp.reference) then
-                goto nextModule
-            end
-            if fader.isRunning(moduleName) then
-                trackInfo.text = string.format("%s: %s", moduleName, messages.fadeInProgress)
-                goto nextModule
-            end
+    if not mp then return end
+    for moduleName in pairs(moduleData) do
 
-            sc.volumeTableDefault = defaults.volumes.modules[moduleName]
-            sc.volumeTableCurrent = this.config.volumes.modules[moduleName]
-
-            if this.cell.isInterior
-            and (moduleName ~= "interiorWeather")
-            and (moduleName ~= "interior") then
-                sc.sliderType = sliderCoefficient
-            else
-                if (ref ~= mp.reference) and (moduleName == "rainOnStatics") then
-                    trackInfo.text = string.format("%s: %s", moduleName, messages.findOutdoorShelter)
-                    trackInfo:register(tes3.uiEvent.help, function(e)
-                        local tooltip = tes3ui.createTooltipMenu()
-                        local tip = messages.findOutdoorShelterTooltip
-                        tooltip:createLabel{ text = tip }
-                    end)
-                    goto nextModule
-                end
-                if cellData.playerUnderwater then
-                    configKey = "und"
-                    sc.sliderType = sliderCoefficient
-                else
-                    configKey = "volume"
-                    sc.sliderType = sliderPercent
+        -- This one has a special regime, doesn't play on any reference
```

Comment:
Just curious - why not play it on the static ref and expose it like windoors here?

Reply:
Game engine weather tracks play unattached. `shelterWeather` only kicks in for specific weathers (4, 5, 6, 7 and 9), when weather tracks are already playing. In the case of rainy weathers, that track is `rainLoopSound`. There's no need to attach them.

Reply:
Silly me. Makes sense!